### PR TITLE
add "pull" action to docker registry client context

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1662088

cc @soltysh @smarterclayton

### CODE REVIEW

**Comment by smarterclayton** [comment]
I was pretty sure that pull was always added.  Did that get lost during a refactor?

File: pkg/oc/cli/image/append/append.go [hunk 1]
```diff
@@ -212,9 +213,9 @@ func (o *AppendImageOptions) Run() error {
 		return err
 	}
 	creds := dockercredentials.NewLocal()
-	ctx := context.Background()
+	ctx := apirequest.NewContext()
```
Comment (soltysh) [inline pkg/oc/cli/image/append/append.go]: Why this context change?

Comment (juanvallejo) [reply pkg/oc/cli/image/append/append.go]: Wanted parity with [the way image/mirror does it](https://github.com/openshift/origin/blob/master/pkg/oc/cli/image/mirror/mirror.go#L239).

**Comment by juanvallejo** [comment]
@smarterclayton it looks like we only add "pull" at the registryclient level if no other actions were specified: https://github.com/openshift/origin/blob/master/pkg/image/registryclient/client.go#L311

**Pull request merged**